### DISCUSSION TITLE
Active menu in navbar doesn't highlight subsections / child pages

### DISCUSSION THREAD

**Discussion by jollydodo** [thuliteio/doks#1178] [Ideas]
## Description 

Minor change and not necessarily an issue/bug, but something that is normally different and would be a visual improvement.  The navbar shows which section is active. This works fine, however, it does not take into account subsections. E.g. if I go to about and then 'about us', neither the parent nor the child have a highlighted <a> element. This is also the case for blog posts for example. 'blog' only stays highlighted on the blog index and not on a single post.

On one of my previous websites I used the default Hugo method ".isMenuCurrent' to highlight the selected link. This method also supports ancestor/child pages out of the box with very simple code. I looked at the Doks implementation and this seems a bit more intricate. 

Default Hugo approach: 

{{ $currentPage := . }}
{{ range site.Menus.main }}
  {{ if $currentPage.IsMenuCurrent .Menu . }}
    <a class="active" aria-current="page" href="{{ .URL }}">{{ .Name }}</a>
  {{ else if $currentPage.HasMenuCurrent .Menu . }}
    <a class="ancestor" aria-current="true" href="{{ .URL }}">{{ .Name }}</a>
  {{ else }}
    <a href="{{ .URL }}">{{ .Name }}</a>
  {{ end }}
{{ end }}

### Visualization
<img width="394" alt="Scherm­afbeelding 2024-01-16 om 22 56 13" src="https://github.com/gethyas/doks/assets/30504008/326e82f8-b340-4725-8087-997bed2cfaf3">

Picture of another Hugo site: stays highlighted even in a post that is part of 'portfolio' using default Hugo implementation. This tells the user that they are currently viewing an item that is part of 'portfolio'. 

Doks: ( "open-inloop" is a child of "over")

<img width="336" alt="Scherm­afbeelding 2024-01-16 om 22 56 35" src="https://github.com/gethyas/doks/assets/30504008/3bcdc64a-7fc4-4a11-bcc2-f17fe9f04211">

Neither 'open-inloop' or 'over' is highlighted as active. 

Preferred outcome:
<img width="203" alt="Scherm­afbeelding 2024-01-16 om 22 56 43" src="https://github.com/gethyas/doks/assets/30504008/b9026a5c-504d-4f15-a390-a9cc54d9885b">

This would make the appearance of the navbar more consistent, because it already works like this for the default 'docs' pages: 
<img width="134" alt="Scherm­afbeelding 2024-01-16 om 23 08 34" src="https://github.com/gethyas/doks/assets/30504008/70ef3889-0d27-4c83-9605-a0c0d6c9e00f">

**Comment by h-enk** [comment]
Thanks! I've added https://github.com/gethyas/doks-core/issues/90
